internal/mode/static/handler.go

```diff
@@ -174,15 +175,34 @@ func (h *eventHandlerImpl) handleEvent(ctx context.Context, logger logr.Logger,
 	}
 }
 
-func (h *eventHandlerImpl) updateNginx(ctx context.Context, conf dataplane.Configuration) error {
-	files := h.cfg.generator.Generate(conf)
+func (h *eventHandlerImpl) updateNginx(
```

Comment:
I gotta get this pattern stuck in my head better.

Reply:
The switch statement in HandleEventBatch is a little ugly, with some duplicates depending on the case. What do you think about 3 functions, `updateNginxConf`, `updateNginxUpstreams`, and `updateNginx`, where the latter makes the decision of which functions to call based on the changeType. It separate the actual updates into their own functions, though it still follow the pattern that you are suggesting we change.

The alternative is a switch like:

case NoChange:
   return
case EndpointsOnly:
   if plus; updateUpstreams
   else update conf
case ClusterChange:
   updateConf
   if plus; updateUpstreams

They're just different enough to not be combined into a single case, but still require duplicate-ish calls, making the switch statement a bit long.

Reply:
Per some of the other conversations about updating the conf, it may even end up looking like:

```
case NoChange:
   return
case EndpointsOnly:
   // Generate Files, write Files, if plus call API, else reload
case ClusterChange:
  // Generate Files, writes files, reload
```

Almost the exact same workflow, except for two conditionals (if plus and if EndpointsOnly). There could be two functions, `GenerateAndWriteFiles` and `UpdateNginx`, though I do like the cohesiveness of having the generation and writing of files all under the "updateNginx" umbrella, since it's directly related.

Reply:
Nevermind, there is a little duplication but I think I can make this cleaner.